server/test/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceTest.java

```diff
@@ -0,0 +1,119 @@
+        ImmutableMap.Builder<OutOfBandManagement.Option, String> builder = new ImmutableMap.Builder<>();
+        builder.put(OutOfBandManagement.Option.ADDRESS, "localhost");
+        builder.put(OutOfBandManagement.Option.DRIVER, "ipmitool");
+        return builder.build();
+    }
+
+    @Test
+    public void testUpdateOutOfBandManagementConfigValid() {
+        OutOfBandManagement config = new OutOfBandManagementVO(123L);
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
+        config = oobmService.updateConfig(config, buildRandomOptionsMap());
+        Assert.assertEquals(config.getAddress(), "localhost");
+        Assert.assertEquals(config.getDriver(), "ipmitool");
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
+    }
+
+    @Test
+    public void testUpdateOutOfBandManagementConfigInValid() {
+        try {
+            oobmService.updateConfig(null, buildRandomOptionsMap());
+            Assert.fail("CloudRuntimeException was expect for out-of-band management not configured for the host");
+        } catch (CloudRuntimeException ignored) {
+        }
+
+        try {
+            oobmService.updateConfig(null, null);
+            Assert.fail("CloudRuntimeException was expect for out-of-band management not configured for the host");
+        } catch (CloudRuntimeException ignored) {
+        }
+
+        OutOfBandManagement config = new OutOfBandManagementVO(123L);
+        config.setAddress(null);
+        config = oobmService.updateConfig(config, null);
+        Assert.assertEquals(config.getAddress(), null);
+        Assert.assertEquals(config.getPowerState(), OutOfBandManagement.PowerState.Disabled);
+    }
+
+    @Test
+    public void testGetOutOfBandManagementOptionsValid() {
+        OutOfBandManagement configEmpty = new OutOfBandManagementVO(123L);
+        ImmutableMap<OutOfBandManagement.Option, String> optionsEmpty = oobmService.getOptions(configEmpty);
+        Assert.assertEquals(optionsEmpty.size(), 0);
+
+        OutOfBandManagement config = new OutOfBandManagementVO(123L);
+        config.setAddress("localhost");
+        config.setDriver("ipmitool");
+        config.setPort(1234);
+        ImmutableMap<OutOfBandManagement.Option, String> options = oobmService.getOptions(config);
+        Assert.assertEquals(options.get(OutOfBandManagement.Option.ADDRESS), "localhost");
+        Assert.assertEquals(options.get(OutOfBandManagement.Option.DRIVER), "ipmitool");
+        Assert.assertEquals(options.get(OutOfBandManagement.Option.PORT), "1234");
+    }
+
+    @Test
+    public void testGetOutOfBandManagementOptionsInvalid() {
+        try {
+            oobmService.getOptions(null);
+            Assert.fail("CloudRuntimeException was expected for finding options of host with out-of-band management configuration");
+        } catch (CloudRuntimeException e) {
+        }
```

Comment:
This ``try`` block and ``fail`` construct is unnecessary.  Changing ``@Test`` to ``@Test(expected = CloudRuntimeException.class)`` accomplishes the same thing.